internal/discoverer/discoverer.go

```diff
@@ -0,0 +1,87 @@
+package discoverer
+
+import (
+	"errors"
+	"fmt"
+	"github.com/api7/apisix-seed/internal/utils"
+	"strconv"
+)
+
+var (
+	queryChecks = [3]string{"event", "entity", "service"}
+	Discoveries = make(map[string]Discover)
+)
+
+type Discover func(disConfig interface{}) Discoverer
+
+// Discoverer defines the component that interact nacos, consul and so on
+type Discoverer interface {
+	Stop()
+	Query(utils.Message) error
+	Watch() chan utils.Message
+}
+
+type Node struct {
+	host   string
+	weight int
+}
+
+type Service struct {
+	name     string
+	nodes    []Node
+	entities []string
```

Comment:
Better to add a comment about the difference between nodes & entities?

Reply:
Done.